src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/history.html.twig

```diff
@@ -38,13 +38,16 @@
         {% if status.employeeLastName is not empty %}
           {{ status.employeeFirstName ~ ' ' ~ status.employeeLastName }}
         {% endif %}
+        {% if status.apiClientId is not empty %}
+          {{ 'Api Client: %api_client_id%'|trans({'%api_client_id%': status.apiClientId}, 'Admin.Global') }}
```

Comment:
Could we get the information about the given id_order_history? What if we would do:

` $this->mutationTracker->addMutationForModule('order_history', (int) $history->id, MutationAction::CREATE);`

I would have to hook into the displayAdminOrderSomething and create my own list of changes to the history, right?

Reply:
So, instead of joining on API_CLIENT, you could only get the information about a given Mutation row for a given history record and return the information accordingly. It could be api_client, module, etc.

Reply:
Ah you want to also display if a modifiaction was made by a module
This was not implemented so far, right now the CQRS query has been modified to return the API client relation, it is done explicitly as CQRS query results must be explicit in their structure, it doesb't return the mutation directly it bases its SQL query on the table and reformat it in a domain way

TLDR; if we want to handle this it should be done in a dedicated PR/new feature

Reply:
Yeah, this would need to have something like this:

```
{% if status.mutationTrackerDetails is not empty %}
{{ status.mutationTrackerDetails }}
{% endif %}
```

```
 LEFT JOIN `' . _DB_PREFIX_ . 'mutation` m ON m.`mutation_table` = "order_history" AND m.`mutation_row_id` = oh.`id_order_history` 
 ```

And we would need to have something to render information based on `mutation.id` if we are keeping the ID anyway.

Formatter could be inside `GetOrderForViewingHandler`

Reply:
Thanks to this, instead of informing about API changes, you're providing the core with the information on the actor who made the mutation. Module, API, employee, doesn't matter.